cmd/genji/shell/shell.go

```diff
@@ -256,9 +257,35 @@ func (sh *Shell) getDB() (*genji.DB, error) {
 		os.Exit(2)
 	}
 
+	err = sh.runPipedInput()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(2)
+	}
+
 	return sh.db, nil
 }
 
+func (sh *Shell) runPipedInput() error {
+	// Check if there is any input being piped in from the terminal
+	stat, _ := os.Stdin.Stat()
+	m := stat.Mode()
+
+	if (m&os.ModeNamedPipe) == 0 /*cat a.txt| prog*/ && !m.IsRegular() /*prog < a.txt*/ { // No input from terminal
+		return nil
+	}
+	data, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		return fmt.Errorf("Unable to read piped input: %w", err)
+	}
+	err = sh.db.Exec(string(data))
```

Comment:
Since data may contain a SELECT, we should use `sh.db.Query` instead:

```go
st, err := sh.db.Query(st)
if err != nil {
  ...
}
defer st.Close()
return record.IteratorToJSON(os.Stdout, st)
```